packages/itwin/grouping-mapping-widget/src/components/customUI/GroupQueryBuilderCustomUI.tsx

```diff
@@ -74,7 +74,11 @@ export const GroupQueryBuilderCustomUI = ({ updateQuery, isUpdating, resetView }
   const onClickResetButton = async () => {
```

Comment:
This should be changed to be `useCallback` with dependencies on `iModelConnection` and `currentPropertyList`.

Reply:
I can somewhat see that `resetView` being wrapped in a `useMutation` with the `onError` being used to console log.

Reply:
Please attempt to do this. It will make the code a bit cleaner. 
```
const { mutate: resetMutation} = useMutation(resetView, {
  onError: (error) => console.error(error),
});
```
Is what I had in mind. Let me know if it works.

Reply:
I've wrapped `resetView` in `useMutation` as suggested, using `onError` to log any errors. The `onClickResetButton` function now calls `resetMutation.mutate()` directly, simplifying the error handling and making the code cleaner.

Reply:
Callback is no longer `async`.

Reply:
I've removed async from onClickResetButton since useMutation handles the async behavior, and the callback is no longer asynchronous. Thanks for catching that!